### ISSUE TITLE
作者你好，从你的论文中我收获很多知识，这是一个很好的工作，请问我要怎么可视化我的分割结果呢？python tools/train_net.py --config-file configs/CalibNet_R50_50e_50q_320size.yaml --num-gpus 1 --eval-only MODEL.WEIGHTS weights/calibnet_r50_50e.pth OUTPUT_DIR output/eval

### ISSUE THREAD

**Issue by 19970799560** [PJLallen/CalibNet#4]

**Comment by PJLallen** [comment]
Hi, you can refer to my other work "OSFormer" for your visualization, the link is: https://github.com/PJLallen/OSFormer/blob/main/tools/visualize_data.py.
Thank you for your attention.

**Comment by rvncel** [comment]
Have you solved it?

**Comment by 19970799560** [comment]
> Have you solved it?

Alright, I just changed a bit of code to see the segmented mask.

**Comment by rvncel** [comment]
I also downloaded the code of the corresponding visualization according to the author's link, and made some minor changes, but it was not possible to form a visual segmentation mask, can you give me some advice?

**Comment by 19970799560** [comment]
> I also downloaded the code of the corresponding visualization according to the author's link, and made some minor changes, but it was not possible to form a visual segmentation mask, can you give me some advice?

Do you have a wechat account? I can send you my modified code.

**Comment by yinyong14** [comment]
I have, this is my WeChat account yong_yin_14 thank you for your help

**Comment by yinyong14** [comment]
I have, this is my WeChat account yong_yin_14 thank you for your help



---- 回复的原邮件 ----
| 发件人 | ***@***.***> |
| 日期 | 2024年11月27日 14:31 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [PJLallen/CalibNet] 作者你好，从你的论文中我收获很多知识，这是一个很好的工作，请问我要怎么可视化我的分割结果呢？python tools/train_net.py --config-file configs/CalibNet_R50_50e_50q_320size.yaml --num-gpus 1 --eval-only MODEL.WEIGHTS weights/calibnet_r50_50e.pth OUTPUT_DIR output/eval (Issue #4) |

I also downloaded the code of the corresponding visualization according to the author's link, and made some minor changes, but it was not possible to form a visual segmentation mask, can you give me some advice?

Do you have a wechat account? I can send you my modified code.

**Comment by PJLallen** [comment]
You can also refer to our latest work for the code of visualization: [https://github.com/PJLallen/InstanceSAM2Eval/blob/main/SAM/notebooks/sam_bbox_show.py](https://github.com/PJLallen/InstanceSAM2Eval/blob/main/SAM/notebooks/sam_bbox_show.py)

**Comment by ZY-Zeng** [comment]
My modification. It maybe not best but it works.

# Visualization

Download code from https://github.com/PJLallen/OSFormer/blob/main/tools/visualize_data.py and put in the *tools* file

Modify codes

- Replace setup function

```python
# old setup function
def setup(args):
    cfg = get_cfg()
    if args.config_file:
        cfg.merge_from_file(args.config_file)
    cfg.merge_from_list(args.opts)
    cfg.freeze()
    return cfg
```

```python
#  new setup function
def setup(args):
    """
    Create configs and perform basic setups.
    """
    cfg = get_cfg()
    add_calibnet_config(cfg)
    cfg.merge_from_file(args.config_file)
    cfg.merge_from_list(args.opts)
    cfg.freeze()
    default_setup(cfg, args)
    setup_logger(output=cfg.OUTPUT_DIR, distributed_rank=comm.get_rank(), name="calibnet")
    return cfg
```

- Replace DatasetMapper function for DatasetMapperWithBasis function

```python
# old DatasetMapperWithBasis function
mapper = DatasetMapperWithBasis(cfg, True)
```

```python
# new DatasetMapper function
mapper = DatasetMapper(cfg, True)

```

- Modify library

```python
# old library
import argparse
import numpy as np
import os
from itertools import chain
import cv2
import tqdm
from PIL import Image

from detectron2.data import DatasetCatalog, MetadataCatalog, build_detection_train_loader
from detectron2.data import detection_utils as utils
from detectron2.data.build import filter_images_with_few_keypoints
from detectron2.utils.logger import setup_logger
from detectron2.utils.visualizer import Visualizer

from adet.config import get_cfg
from adet.data.dataset_mapper import DatasetMapperWithBasis
```

```python
# new library
from itertools import chain
import cv2
import tqdm
import detectron2.utils.comm as comm

from PIL import Image

from detectron2.data import DatasetCatalog, MetadataCatalog, build_detection_train_loader
from detectron2.data import detection_utils as utils
from detectron2.data.build import filter_images_with_few_keypoints
from detectron2.utils.logger import setup_logger
from detectron2.utils.visualizer import Visualizer
from detectron2.engine import AutogradProfiler, DefaultTrainer, default_argument_parser, default_setup, launch
from detectron2.config import get_cfg
from detectron2.data import MetadataCatalog, build_detection_train_loader, build_detection_test_loader, DatasetMapper

from calibnet import add_calibnet_config, COCOMaskEvaluator
```

**Comment by ycyg8** [comment]
> 我的改造。这可能不是最好的，但确实有效。
> 
> # 可视化
> 从 https://github.com/PJLallen/OSFormer/blob/main/tools/visualize_data.py 下载代码，放进_工具_文件里
> 
> 修改代码
> 
> * 替换设置函数
> 
> # old setup function
> def setup(args):
>     cfg = get_cfg()
>     if args.config_file:
>         cfg.merge_from_file(args.config_file)
>     cfg.merge_from_list(args.opts)
>     cfg.freeze()
>     return cfg
> #  new setup function
> def setup(args):
>     """
>     Create configs and perform basic setups.
>     """
>     cfg = get_cfg()
>     add_calibnet_config(cfg)
>     cfg.merge_from_file(args.config_file)
>     cfg.merge_from_list(args.opts)
>     cfg.freeze()
>     default_setup(cfg, args)
>     setup_logger(output=cfg.OUTPUT_DIR, distributed_rank=comm.get_rank(), name="calibnet")
>     return cfg
> * 将 DatasetMapper 函数替换为 DatasetMapperWithBasis 函数
> 
> # old DatasetMapperWithBasis function
> mapper = DatasetMapperWithBasis(cfg, True)
> # new DatasetMapper function
> mapper = DatasetMapper(cfg, True)
> * 修改库
> 
> # old library
> import argparse
> import numpy as np
> import os
> from itertools import chain
> import cv2
> import tqdm
> from PIL import Image
> 
> from detectron2.data import DatasetCatalog, MetadataCatalog, build_detection_train_loader
> from detectron2.data import detection_utils as utils
> from detectron2.data.build import filter_images_with_few_keypoints
> from detectron2.utils.logger import setup_logger
> from detectron2.utils.visualizer import Visualizer
> 
> from adet.config import get_cfg
> from adet.data.dataset_mapper import DatasetMapperWithBasis
> # new library
> from itertools import chain
> import cv2
> import tqdm
> import detectron2.utils.comm as comm
> 
> from PIL import Image
> 
> from detectron2.data import DatasetCatalog, MetadataCatalog, build_detection_train_loader
> from detectron2.data import detection_utils as utils
> from detectron2.data.build import filter_images_with_few_keypoints
> from detectron2.utils.logger import setup_logger
> from detectron2.utils.visualizer import Visualizer
> from detectron2.engine import AutogradProfiler, DefaultTrainer, default_argument_parser, default_setup, launch
> from detectron2.config import get_cfg
> from detectron2.data import MetadataCatalog, build_detection_train_loader, build_detection_test_loader, DatasetMapper
> 
> from calibnet import add_calibnet_config, COCOMaskEvaluator

Is your code directly drawing images based on the annotation results in the dataset? I see that the code seems to mean this. Can you explain it, or do you have code for masking a single image? I haven't implemented it yet.
